Add attacker model (#59)
tls-attestation/use-cases-and-properties: draft-ietf-seat-use-cases.md

prompt template) can change independently of the binary/image and usually
more frequently than typical platform TCB updates {{AI-agents}}.

# Attacker Model
{: #attacker-model }

This section describes the attack vectors that motivate the security goals in the
following section, and the requirement each imposes on a conforming solution.

> 

Reviewer: I suggest adding, as a preamble:

> We assume the Internet Threat Model of RFC 3552, in which an attacker can observe, modify, delete, and inject packets on the communication path.

## Cross-Connection and Cross-Peer Evidence Reuse
{: #evidence-reuse }

An adversary attempts to present Evidence outside the specific interaction for
which it was produced:

> * Cross-connection replay: Evidence obtained from a Target Environment in one

Reviewer 2: ```suggestion
* Cross-connection replay: Evidence obtained from an Attesting Environment in one
```

  connection is replayed in a later, separate connection, where it no longer
  reflects the current state.

> * Cross-peer relay: a peer obtains Evidence from a Target Environment and

Reviewer 2: I find the description of "Cross-peer relay" hard to read.

Author: Has been changed

>   passes it to a different, colluding peer, which injects it into its own

Reviewer: I have a problem with "colluding peer" here. Even standard TLS is not secure in the face of similar collusion. If server S1 is colluding with server S2 then both can pretend to be one another etc.

  handshake with a Relying Party.

If Evidence is not uniquely tied to a single TLS connection, an adversary can
use old Evidence in a different connection or share it with unauthorized entities, 
leading the Relying Party to trust an outdated or unverified environment.

> ## Key Substitution

Reviewer 2: It was hard to read for me. A figure would help.

Author: Is it still the case after the rewrite?

A peer presents valid Evidence from a Target Environment together with
the private key it uses to authenticate the TLS connection, but that private key
was not generated or protected within the attested environment.
When the Evidence and the authentication key are appraised independently,
the Relying Party may conclude that the key benefits from the
protection properties asserted in the Evidence when it does not. Proof of
possession does not detect this: the presenter genuinely controls the key, so
the handshake succeeds, while the key may reside in software or in a different
module outside the attested environment.

A malicious peer uses this to have its own key, which is not protected by 
the attested environment, treated by the Relying Party as if it were. 
The Relying Party may then release secrets or sensitive data to it, 
or allow privileged operations, that it would refuse for a software-held key; 
and because the key is not confined, the peer can copy it and use it on 
other hosts, so the attested identity is no longer bound to a 
attested environment.

> ## Evidence Exposure under Compromised Traffic Secrets

Reviewer 2: Which traffic secrets: handshake or application?

Reviewer 3: It would apply to both, right? If the TLS signing key cannot be trusted until after remote attestation completes, in the case an active attacker has compromised the TLS signing key and is exploiting it, then Evidence will be exposed to the attacker even when conveyed during a post-TLS Authenticator handshake —even when otherwise fully encrypted under the derived ephemeral application traffic keys of that session.

Author: Clarified.

Re-attestation may be performed over an long-lived (D)TLS connection. If the
traffic secrets of that connection have been compromised (see
{{I-D.ietf-tls-extended-key-update}} for more details on the threat), an adversary

> holding them can decrypt the re-attestation exchange and read the Evidence.

Reviewer 2: Why "read the Evidence" is a problem?

> Besides leaking platform attestation details, the adversary may attempt to reuse

Reviewer: It's not clear from this text whether we consider "leaking the attestation" at attack.

the captured Evidence on a different connection (see {{evidence-reuse}}).

> ## Posture Drift on Long-Lived and Resumed Connections

Reviewer 2: Clarify "Posture" vs. state?

Author: Changing to state since we already use it in the text underneath instead of posture.

The Relying Party appraises the Attester's Evidence during the TLS handshake. 
Its posture can change afterwards without the connection being re-established. 

Two situations are of particular concern:

* Long-lived connections: the connection stays open while the Target Environment
  changes state, so sensitive data keeps flowing over an environment whose Evidence
  is now stale.

> * Resumption: the Relying Party reconnects using a session ticket, skipping the

Reviewer: There are two very different cases here:

1. The session ticket was generated *after* attestation had been performed. The system's state may have changed since, but at least we know that previously it was secure.
2. The ticket was generated before attestation, all bets are off.

  handshake and the attestation exchange, so communication continues under the new,
  unverified state.

In both cases the Relying Party keeps exchanging data with a Target Environment
whose posture has changed, bypassing RA for ongoing or resumed traffic. This is
related to the exposure discussed under Runtime Attestation.

## RA Negotiation Downgrade

Peers negotiate whether RA is used, the Evidence formats they can produce or
consume, and the attestation models they support. An on-path adversary can

> tamper with this negotiation: stripping the RA capability so the peers fall

Reviewer 2: What is stripping? Use terminology consistent with RFC9846.

Reviewer 3: > What is stripping? Use terminology consistent with RFC9846.

I believe stripping is a commonly understood term of art regarding MiTM threat models. For example, in academic literature associated with TLS, I’ve seen it here, see page 7 https://www.ieee-security.org/TC/SP2013/papers/4977a511.pdf

Author: It has been removed now.

back to a plain secure channel with no attestation, or forcing selection of a
weaker Evidence format or attestation model than both peers would otherwise
support. Because this negotiation precedes any established trust in the Target
Environment, an adversary that can modify these messages may silently remove or
degrade attestation without either peer detecting the change. This is the

> adversarial counterpart to the graceful fallback under the Negotiation and

Reviewer 3: I think graceful fallback for remote attestation does defeat the point. If a client asks for remote attestation and does not receive it, it should not proceed with the connection, since the endpoint refused (for whatever reason) to attest. That’s not feedback for this PR in particular other than affirming the note is useful in contrast to what the document already specifies elsewhere. But I suppose that may be an implementation detail?

Reviewer 4: Good point, I propose to fix the last line as follows:

This is the adversarial counterpart to the graceful fallback under the Negotiation and Capability Discovery goal; the end goal is that a Relying Party requiring attestation rejects a peer that does not attest.

> Capability Discovery goal.

Reviewer 2: Add a reference to that section.

baseline.

Unless stated otherwise, the Relying Party is not compromised and correctly
performs the checks required by the protocol and its appraisal policy. The
Attester is referenced in terms of its Attesting and Target Environments, as
described in {{Section 3.1 of -rats-arch}}. The Attesting Environment is the
entity responsible for using the Attestation Key to sign Evidence for the
Target Environment.

The baseline security assumptions of the TLS v1.3 protocol apply. See
{{Appendix F of -tls}} for the relevant properties.

The TLS stack of the attester is assumed to be running within the TEE.

## Attacker Capabilities

The attacks in this section use the following attacker profiles:

* **Network attacker:** Controls communication between the peers and can
  observe, inject, modify, drop, delay, reorder, and replay messages. This
  attacker does not control either endpoint or possess their cryptographic
  secrets. This closely follows the attacker model of {{-int-threat-model}}.
* **Malicious peer:** Controls a protocol endpoint and can deviate arbitrarily
  from the protocol. Control of the peer does not, by itself, imply compromise

>   of an Attesting Environment or its Attestation Key.

Reviewer 4: Malicious peer: One TLS peer may be compromised and may deviate arbitrarily from the protocol. The other peer, including its Attesting Environment and Attestation Key, is assumed to remain uncompromised.

Reviewer 3: This one in particular has taken a couple of rounds of re-reading to try and be sure I fully understand it. I think the word 'peer' is hitting semantic oversaturation for me, in this particular case.

Is this meant to say that a "malicious peer" may be either one or the other the attesting instance or a spoofed RP, just as likely? Even trying to rephrase it in my own words makes me feel like I'm struggling. So your own phrasing may be fine.

Author: I think this is trying to say that the entity trying to prove itself through remote attestation may be malicious, but that doesn't imply they have full control over an attesting environment or its attestation key (to sign arbitrary evidence, basically).

@tireddy2 - I don't think your proposed text has the same properties in mind. You seem to imply that the malicious peer controls everything on its side, including the RoT of the attester, but the other party is untouched.

How about:

> Malicious peer: Controls a TLS endpoint and can deviate arbitrarily from the protocol. The TLS endpoint is NOT running in an environment that can produce Evidence that is acceptable to the Relying Party. Control of this endpoint also does not, by itself, imply compromise of any Attesting Environment or its Attestation Key.

Reviewer 4: > I think this is trying to say that the entity trying to prove itself through remote attestation may be malicious, but that doesn't imply they have full control over an attesting environment or its attestation key (to sign arbitrary evidence, basically).
> 
> @tireddy2 - I don't think your proposed text has the same properties in mind. You seem to imply that the malicious peer controls everything on its side, including the RoT of the attester, but the other party is untouched.

No, I meant that only one TLS peer is malicious and can deviate arbitrarily from the protocol, but it does not control or compromise the Attesting Environment or its Attestation Key. The other TLS peer, including its Attesting Environment and Attestation Key, remains uncompromised.

> 
> How about:
> 
> > Malicious peer: Controls a TLS endpoint and can deviate arbitrarily from the protocol. The TLS endpoint is NOT running in an environment that can produce Evidence that is acceptable to the Relying Party. Control of this endpoint also does not, by itself, imply compromise of any Attesting Environment or its Attestation Key.

Malicious peer: Controls a TLS endpoint and can deviate arbitrarily from the protocol. The TLS endpoint is not running in an environment that can produce Evidence acceptable to the Relying Party. The attacker does not control or compromise the Attesting Environment or its Attestation Key.

**Attacker capabilities:** A malicious peer retains valid Evidence generated
earlier in the current TLS connection.

**Targeted assurance:** Evidence presented in response to a re-attestation
request reflects the current state of the Target Environment.

**Preconditions and attack:** Evidence is bound to the TLS connection but not
uniquely to an individual attestation exchange. After valid Evidence is accepted,
the state of the Target Environment changes and the Relying Party requests new
Evidence. The malicious peer retransmits the earlier Evidence in new TLS records,
causing the Relying Party to treat the earlier state as current.

**Why validation succeeds:** The Evidence remains authentic and its
connection-level binder still matches. TLS record replay protection does not
detect the attack because the old Evidence object is retransmitted as new
application data.

**Required invariant:** Every Evidence response is fresh for a particular
attestation exchange and is bound to a unique challenge or equivalent freshness
mechanism. Evidence accepted for one attestation exchange cannot satisfy a later
request on the same TLS connection.

### Evidence Relay

> **Attacker capabilities:** A malicious peer can establish a TLS connection with

Reviewer 4: Attacker capabilities: The attacker, acting as a malicious TLS peer, can establish a TLS connection with a Relying Party and can obtain Evidence about a separate Target Environment that contains the correct binder for the attacker's connection as its challenge value. How the attacker obtains such Evidence is not constrained by this model. The attacker does not need to control or collude with the Attesting or Target Environment.

a Relying Party and can obtain Evidence about a separate Target Environment that
contains the correct binder for the attacker's connection as its challenge
value. How the attacker obtains such Evidence is not constrained by this model.
The attacker does not need to control or collude with the Attesting or Target
Environment.

> **Targeted assurance:** The Evidence describes the Target Environment

Reviewer 4: Targeted assurance: The Relying Party incorrectly assumes that the authentication private key has the key protection properties asserted by the Evidence.

Author: I can add the part about the authentication private key to the assurance, but the general point about the attested endpoint is needed regardless, because the appraisal is not strictly about the key.

Reviewer 4: Agreed. 
Targeted assurance: The Relying Party incorrectly attributes the claims asserted by the Evidence to the TLS endpoint and assumes that the authentication private key has the protection properties asserted by the Evidence.

associated with the peer and connection being appraised by the Relying Party.

**Preconditions and attack:** The attacker obtains the binder for its TLS
connection with the Relying Party. It then obtains authentic, fresh Evidence
about a separate Target Environment in which that binder appears as the
challenge value and presents the Evidence to the Relying Party. If the Evidence
is accepted, the Relying Party attributes the state of the separate Target
Environment to the attacker's endpoint and TLS connection.

> **Why validation succeeds:** The Evidence is authentic and fresh, and its

Reviewer 4: Why validation succeeds: Proof of possession establishes that the peer controls the private key, and Evidence establishes properties of the Target Environment. Neither check establishes that this particular private key was generated, is stored, and is used strictly within the Attesting Environment.

Author: This might be correct in some deployments, but as a general statement the mechanical check that fails is for the actual binding between attestation and TLS.

You can still have relay even if you have key attestation. For example, if the TEE is both an attestation oracle and a signing oracle, it can give you both valid evidence proving that the key is secure, and a signature with that key for whatever the malicious TLS endpoint wants (like a CertificateVerify). The underlying problem is that the evidence is in no way bound to the execution of the TLS peer using it. This is also the counterargument for the comment above regarding the assurance.

Reviewer 4: I agree that the key issue is binding the Evidence to the TLS execution. However, I don't think we should assume that a TEE can act as a signing oracle. We should make that an explicit assumption if it is required for the attack.

Author: > However, I don't think we should assume that a TEE can act as a signing oracle. 

I think that depends on the relationships within the system, and what we want to consider as valid. I agree that if the RP and TEE-owner are part of the same organisation, or within the same trust boundary, then it doesn't make sense for the TEE to do random stuff.

But if the TEE-owner and RP are mutually-distrusting, then the RP has to be a lot more thorough. A malicious TEE-owner can run anything they like in the TEE. They might want to pretend they're running some specific service inside, while only using it for the oracles I mentioned. It's up to the RP to determine what's valid.

So my question is: do we want to restrict this to situations where the TEE-owner is trustworthy by definition?

Reviewer 4: I think the trustworthiness of the TEE owner is ultimately determined by the verifier's appraisal policy. The protocol should not assume that the TEE owner is trustworthy by definition.

Author: I agree, but then this supports my earlier statement that just proving the key is protected is not enough, because the "attestation + signing oracle" attester passes that assessment. The verifier has to establish whether the attester's TEE is the one running the TLS stack or not. That can't be done just by binding to or attesting the CertificateVerify-signing key, because those can be relayed too.

Author: Hmm, I think I get where the confusion is now. I think your "key attestation" requirement subsumes my requirement, implicitly: you're requiring proof not just that the key was created in this TEE and is not exported, but that it can only be used for TLS CertificateVerify signing for this protocol. If that holds, there's no possibility for relay, I agree, but we should make this second part clear too.

Reviewer 4: > Hmm, I think I get where the confusion is now. I think your "key attestation" requirement subsumes my requirement, implicitly: you're requiring proof not just that the key was created in this TEE and is not exported, but that it can only be used for TLS CertificateVerify signing for this protocol. If that holds, there's no possibility for relay, I agree, but we should make this second part clear too.

Agreed.

and those assertions are appraised together with the connection authentication.

## Evidence Privacy Loss

### Under Ephemeral-Key Compromise

**Attacker capabilities:** An ephemeral-key attacker possesses an endpoint's
ephemeral key-establishment secret for the handshake and observes the handshake
transcript, including the peer's public key share.

**Targeted assurance:** Evidence conveyed during or after the handshake remains
confidential from parties other than the connection endpoints.

**Preconditions and attack:** The attacker records the handshake and protected
records carrying Evidence. Using the compromised ephemeral private key, the
peer's public key share, and the recorded handshake messages, the attacker
computes the shared secret and derives the handshake and application traffic
secrets. It can then decrypt Evidence protected under those secrets, including
Evidence decrypted retrospectively from recorded traffic after the compromise.

**Why validation succeeds:** This is a secure-channel confidentiality failure,
not an Evidence-validation failure. The Evidence can remain authentic, fresh,
and correctly bound to the connection even though its contents are disclosed.

> **Required invariant:** Ephemeral key-establishment secrets are protected and

Reviewer 4: Required invariant: Ephemeral key-establishment secrets are protected and erased when no longer needed. A design that claims recovery from compromise of such a secret provides Post-Compromise Security (PCS) by establishing new traffic secrets using fresh key-establishment input independent of the compromised secret before sending further Evidence.

prompt template) can change independently of the binary/image and usually
more frequently than typical platform TCB updates {{AI-agents}}.

# Attacker Model
{: #attacker-model }

This section defines the attacker capabilities and attack scenarios that a
solution integrating RA with a secure channel needs to consider. Attacks
in-scope for IETF protocols generally assume the Internet Threat Model of
{{-int-threat-model}}. This corresponds to the "network attacker" described
below. Some scenarios explicitly grant the attacker capabilities beyond that
baseline.

Unless stated otherwise, the Relying Party is not compromised and correctly
performs the checks required by the protocol and its appraisal policy. The
Attester is referenced in terms of its Attesting and Target Environments, as
described in {{Section 3.1 of -rats-arch}}. The Attesting Environment is the
entity responsible for using the Attestation Key to sign Evidence for the
Target Environment.

> The baseline security assumptions of the TLS v1.3 protocol apply. See

Reviewer 2: This has started to collude the threat model with the properties.

{{Appendix F of -tls}} for the relevant properties.

The TLS stack of the attester is assumed to be running within the TEE.

## Attacker Capabilities

The attacks in this section use the following attacker profiles:

* **Network attacker:** Controls communication between the peers and can
  observe, inject, modify, drop, delay, reorder, and replay messages. This
  attacker does not control either endpoint or possess their cryptographic
  secrets. This closely follows the attacker model of {{-int-threat-model}}.
* **Malicious peer:** Controls a TLS endpoint and can deviate arbitrarily from

>   the protocol. The TLS endpoint is not running in an environment that can

Reviewer 2: I don't understand this sentence with 'not.' Why can't you say it without negation? Is it running in Target Environment? Something like: 

The TLS endpoint is running in Target Environment.

# Attacker Model
{: #attacker-model }

This section defines the attacker capabilities and attack scenarios that a
solution integrating RA with a secure channel needs to consider. The baseline

> attacker follows the Internet Threat Model of {{-int-threat-model}}. Some

Reviewer 2: This does not seem to align with [this](https://github.com/ietf-wg-seat/draft-ietf-seat-use-cases/pull/59#issuecomment-5385942401). "Internet Threat Model" contains threat model in it.

Author: Right, but the Internet Threat Model assumes a very specific attacker, with very specific capabilities. The first few paragraphs of [section 3 of RFC3552](https://www.rfc-editor.org/info/rfc3552/#section-3) (up to sec. 3.1) describe that attacker. What I think this is saying is that the "baseline" attacker ("network attacker" underneath) has the capabilities assumed by that threat model. I'll try to make this clearer.

Author: Please have a look again and see if the current text makes it clearer!
